regolith/main.py

```diff
@@ -349,6 +349,7 @@ def main(args=None):
             dbs = commands.helper_db_check(rc)
         with connect(rc, dbs=dbs) as rc.client:
             CONNECTED_COMMANDS[rc.cmd](rc)
+        commands.google_calendar_updater(rc)
```

Comment:
I am pretty sure this is the wrong pattern, but we may need to discuss it if there is some reason it has to be this way.

Reply:
I put the google calendar function in here, because I wanted it to be outside of the context manager. That way, the program will dump the presentation files first without having to wait for the google calendar user consent. Is it better to have this function in a_presentationhelper.py?

Reply:
This is a pretty major perturbation of the whole structure of the Regolith package, just as a workaround for a small annoyance.  It breaks all the logic will be hard to maintain in the future because we won't know why it is there breaking the logic, so I dislike it very much, though I understand your motivation.   It probably needs more discussion and a better solution.  I am guessing that there is a better solution than this if we can put our thinking caps on .

In general, we want our code to structured to be understandable and maintainable and there has to be a VERY good reason to do something that breaks logic.

Reply:
can you discuss it maybe with Zach and Connor?

Reply:
Sure, I'll reach out to them